src/bp/ui-studio/src/web/views/FlowBuilder/nodeProps/ActionDialog.tsx

```diff
@@ -0,0 +1,167 @@
+import { Button, Dialog, FormGroup, HTMLSelect, Label, NonIdealState } from '@blueprintjs/core'
+import axios from 'axios'
+import { ActionParameterDefinition, ActionServersWithActions } from 'common/typings'
+import _ from 'lodash'
+import React, { FC, useEffect, useState } from 'react'
+
+import { Action, Parameters } from '../diagram/nodes_v2/ActionNode'
+
+import style from './style.scss'
+import { ActionParameters } from './ActionParameters'
+export interface ParameterValue {
+  definition: ActionParameterDefinition
+  value: string
+}
+
+interface ActionDialogProps {
+  name: string
+  parameters: Parameters
+  actionServerId: string
+  isOpen: boolean
+  onClose: () => void
+  onSave: (action: Action) => void
+}
+
+const ActionDialog: FC<ActionDialogProps> = props => {
+  const { isOpen, onClose, onSave } = props
+
+  const [actionServers, setActionServers] = useState<ActionServersWithActions[]>([])
+  const [name, setName] = useState(props.name)
+  const [parameters, setParameters] = useState(props.parameters)
+  const [actionServerId, setActionServerId] = useState(props.actionServerId)
+
+  useEffect(() => {
+    const fetchActionServers = async () => {
+      const response = await axios.get(`${window.BOT_API_PATH}/actionServers`)
+      setActionServers(response.data)
+    }
+
+    // tslint:disable-next-line: no-floating-promises
+    fetchActionServers()
+  }, [])
+
+  if (actionServers.length === 0) {
+    return null
+  }
+
+  const actionIsValid = () => {
+    return !!name && !!actionServerId
+  }
+
+  const isNewNode = name === '' && _.isEmpty(parameters) && actionServerId === ''
+  if (isNewNode) {
+    const defaultActionServer = actionServers[0]
+    const defaultAction = defaultActionServer.actions[0]
+
+    const defaultName = defaultAction?.name
+    const defaultParameters = {}
+    const defaultActionServerId = defaultActionServer.id
+
+    setName(defaultName)
+    setParameters(defaultParameters)
+    setActionServerId(defaultActionServerId)
+    if (actionIsValid()) {
+      onSave({ name: defaultName, parameters: defaultParameters, actionServerId: defaultActionServerId })
+    }
+    return null
```

Comment:
There's a lot of placeholder variables, you don't need the defaultName, defaultParameters and defaultActionServerId. You can just set them directly, then you don't even need to change the names in your onSave